node/mock-task.ts

```diff
@@ -64,6 +62,32 @@ module.exports.getBoolInput = task.getBoolInput;
 module.exports.getDelimitedInput = task.getDelimitedInput;
 module.exports.filePathSupplied = task.filePathSupplied;
 
+export function getVariable(name: string): string | undefined  {
```

Comment:
I'm *not* normalizing the variable name here (upper casing, replacing periods). It's not clear to me if you want all that business logic when you're providing mock answers.

Reply:
I think we should be doing this - we don't want our tests behaving one way (case sensitive, etc...) while the actual agent behaves differently (non-casesensitive, etc...)

Reply:
I added normalization. I mocked `setVariable` as well, because this should probably work when mocking:

```typescript
setVariable("foo", "bar");
const val = getVariable("foo");
```

What haven't done yet is throw on multi-line secrets. If we need to do that too, I'm starting to wonder if we should just have our tests, or mock-answer.ts, call the real `setVariable`. Perhaps we can refactor the variable code so it doesn't access `process.env` when running unit tests.

Reply:
Added error on multi-line secrets. I may take a look at mocking just the access to `process.env` and `im.vault`.

Reply:
I'm looking into the feasibility of adding real and mock backing stores so we can share logic across task and mock-task.

interfaces.ts (new file)
```ts
export interface SecretStore { // implemented by Vault and some test class.
    storeSecret(name: string, data: string): boolean;
    retrieveSecret(name: string): string | undefined;
}
```

internal.ts
```ts
// export var _vault: vm.Vault;
export var _vault: iface.SecretStore;
```